demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -215,14 +362,423 @@ static bool getS3ObjectFileSize( size_t * pFileSize,
+ * #JSONBadParameter if any of the response parameters that needs to be parsed are empty;
+ * #JSONNotFound if the key to be parsed is not in the response.
+ */
+static JSONStatus_t parseCredentials( HTTPResponse_t * response,
+                                      SigV4Credentials_t * sigvCreds );
+
+/**
+ * @brief Retrieve the temporary credentials from AWS IOT Credential Provider.
+ *
+ * @param[in] pTransportInterface The transport interface for making network
+ * @param[in] pDateISO8601Len Length of the buffer provided to store ISO8601 formatted date.
+ * @param[in,out] response Response buffer to store the HTTP response received.
+ * @param[out] pDateISO8601 Buffer to store the ISO8601 formatted date.
+ * @param[out] sigvCreds Buffer to store the parsed credentials.
+ *
+ * @return `true` if credentials are retrieved successfully otherwise 'false`.
+ */
+static bool getTemporaryCredentials( TransportInterface_t * transportInterface,
+                                     size_t pDateISO8601Len,
+                                     HTTPResponse_t * response,
+                                     char * pDateISO8601,
+                                     SigV4Credentials_t * sigvCreds );
+
+/**
+ * @brief Hex digest of provided string parameter.
+ *
+ * @param[in] pInputStr Input String to encode.
+ * @param[in] inputStrLen Length of Input String to encode.
+ * @param[out] pHexOutput Hex representation of @p pInputStr.
+ */
+static void lowercaseHexEncode( const char * pInputStr,
+                                size_t inputStrLen,
+                                char * pHexOutput );
+
+/**
+ * @brief Calculate SHA256 digest.
+ *
+ * @param[in] pInput Input string to hash.
+ * @param[in] ilen Length of input string.
+ * @param[out] pOutput Buffer to store the generated hash.
+ */
+static int32_t sha256( const char * pInput,
+                       size_t ilen,
+                       char * pOutput );
+
+/**
+ * @brief Application-defined Hash Initialization function provided
+ * to the SigV4 library.
+ *
+ * @note Refer to SigV4CryptoInterface_t interface documentation for this function.
+ */
+static int32_t sha256Init( void * hashContext );
+
+/**
+ * @brief Application-defined Hash Update function provided to the SigV4 library.
+ *
+ * @note Refer to SigV4CryptoInterface_t interface documentation for this function.
+ */
+static int32_t sha256Update( void * hashContext,
+                             const char * pInput,
+                             size_t inputLen );
+
+/**
+ * @brief Application-defined Hash Final function provided to the SigV4 library.
+ *
+ * @note Refer to SigV4CryptoInterface_t interface documentation for this function.
+ */
+static int32_t sha256Final( void * hashContext,
+                            char * pOutput,
+                            size_t outputLen );
+
+/**
+ * @brief CryptoInterface provided to SigV4 library for generating the hash digest
+ * and signature.
```

Comment:
The crypto interface does not generate signature; it only generates the hash digest
```suggestion
```